frontend/ui/src/app/deployments/deployment-targets.component.ts

```diff
@@ -114,14 +115,23 @@ export class DeploymentTargetsComponent implements AfterViewInit, OnDestroy {
         deploymentTargets
           .map((dt) =>
             dt.deployments.map((d) =>
-              this.getAvailableVersions(d).pipe(map((versions) => ({dt, d, version: this.findMaxVersion(versions)})))
+              this.getAvailableVersions(d).pipe(
+                map((versions) => {
+                  const version = this.findMaxVersion(versions);
+                  if (version) {
+                    return {dt, d, version};
+                  }
+                  return undefined;
+                })
+              )
             )
           )
           .flat()
       )
     ),
+    map((dts) => dts.filter((dt) => !!dt)),
```

Comment:
I've kind of started preferring this more explicit style recently, especially when handling basic types because stuff like `!!'' === false` (which is not the case here but still)… what do you think?

```suggestion
    map((dts) => dts.filter((dt) => dt !== undefined)),
```